grazer/src/main/webapp/WEB-INF/spring/spring-security.xml

```diff
@@ -1,22 +1,22 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <beans:beans xmlns="http://www.springframework.org/schema/security"
-	xmlns:beans="http://www.springframework.org/schema/beans" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-	xmlns:context="http://www.springframework.org/schema/context"
-	xsi:schemaLocation="http://www.springframework.org/schema/beans
-	http://www.springframework.org/schema/beans/spring-beans-3.2.xsd
-	http://www.springframework.org/schema/security
-	http://www.springframework.org/schema/security/spring-security-4.1.xsd">
+    xmlns:beans="http://www.springframework.org/schema/beans" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns:context="http://www.springframework.org/schema/context"
+    xsi:schemaLocation="http://www.springframework.org/schema/beans
+    http://www.springframework.org/schema/beans/spring-beans-3.2.xsd
+    http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context.xsd
+    http://www.springframework.org/schema/security
+    http://www.springframework.org/schema/security/spring-security-4.1.xsd">
+        
+    <beans:bean id="encoder"
+        class="org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder">
+        <beans:constructor-arg name="strength" value="4"></beans:constructor-arg>
+    </beans:bean>
 
-	<beans:bean id="encoder"
-		class="org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder">
-		<beans:constructor-arg name="strength" value="4"></beans:constructor-arg>
-	</beans:bean>
-
-	<authentication-manager alias="authenticationManager"
-		erase-credentials="false">
-		<authentication-provider>
-			<user-service id="userDetailsService" properties="classpath:/user.properties" />
-		</authentication-provider>
-	</authentication-manager>
+    <authentication-manager alias="authenticationManager" erase-credentials="false">
+        <authentication-provider>
+            <user-service id="userDetailsService" properties="classpath:/user.properties" />
+        </authentication-provider>
+    </authentication-manager>
 
 </beans:beans>
```

Comment:
Its best to maintain the security related configurations either in xml file or java file. We do not want the configurations to be spread across multiple files of different format e.g. grazer/src/main/webapp/WEB-INF/spring/spring-security.xml and grazer/src/main/java/edu/asu/diging/grazer/config/SecurityContext.java.

Reply:
The username and password is being retrieved from a properties file, and there is no implementation for that in java config. That's why I have kept both the xml and java file